tests/fixtures/tree-sitter-lint-plugin-replace-foo-with-bar/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+use std::sync::Arc;
+
+use tree_sitter_lint::{rule, violation, Plugin, Rule};
+
+pub fn instantiate() -> Plugin {
+    Plugin {
+        name: "replace-foo-with-bar".to_owned(),
```

Comment:
It's a little bit weird that a plugin declares its name in addition to it being assumed that it can be derived from the crate name (and vice-versa)

I don't remember exactly why I added it explicitly

I guess otherwise it would be in the custom local binary code that it could say create `(String, Plugin)` tuples (of what it thinks/knows the plugin's name is + the `Plugin` instance returned by the plugin crate's entry point) and those would be what would get passed to end up in `Config`'s `.all_plugins`?